### PR TITLE
ss.initail loader added

### PR DESCRIPTION
**Pull request by swetha571**
# Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [ ] The code follows the code style and conventions of this project.
- [ ] The change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.

# Description of the change
Added initial loader with page load and used same loader when the device Group changed accept alarm grid.
...

# Motivation for the change

...

### CODE REVIEW

File: src/actions/kpiActions.js [hunk 1]
```diff
@@ -25,7 +26,11 @@ export const refreshAllChartData = (
       return;
     }
     const deviceIdsCsv = devices.items.map(device => device.Id).join(',');
-    dispatch(indicatorStart('kpi'));
+    if (refreshFlag) {
```
Comment (spryor) [inline src/actions/kpiActions.js]: `dispatch(indicatorStart(refreshFlag ? 'kpi' : 'kpiInitial')); `

File: src/components/deviceMap/deviceMap.js [hunk 2]
```diff
@@ -186,7 +186,8 @@ class DeviceMap extends Component {
 
 const mapStateToProps = state => {
   return {
-    showSpinner : state.indicatorReducer.map,
+    showSpinner : state.indicatorReducer.kpi,
```
Comment (spryor) [inline src/components/deviceMap/deviceMap.js]: There are several places in this PR with spaces before the `:`. I'm not sure how they keep showing up. Are they being added automatically perhaps? Please do a sweep and remove them all.

File: src/components/dashboardPanel/dashboardPanel.js [hunk 3]
```diff
@@ -27,7 +27,9 @@ class DashboardPanel extends Component {
           </div>
           {this.props.actions ? <div className="panel-actions-container">{this.props.actions}</div> : ''}
         </div>
-        <div className="panel-contents">{this.props.children}</div>
+        <div className="panel-contents">
+        {this.props.children}
+        { this.props.initialIndicator ? <div className="initial-loading"><Spinner size='large'/></div> : null}</div>
```
Comment (spryor) [inline src/components/dashboardPanel/dashboardPanel.js]: Return the panel-contents to its previous state and move the loading indicator to be a sibling element to the panel-contents.

File: src/components/dashboardPanel/dashboardPanel.scss [hunk 4]
```diff
@@ -9,6 +9,20 @@ $padding: 15px;
   color: #fff;
   margin-bottom: 16px;
```
Comment (spryor) [inline src/components/dashboardPanel/dashboardPanel.scss]: You need to make dashboard-panel `position: relative`.

File: src/components/dashboardPanel/dashboardPanel.scss [hunk 5]
```diff
@@ -9,6 +9,20 @@ $padding: 15px;
...
+    position: absolute;
+    top: 0px;
+    right: 0px;
+    left: 0px;
+    right: 0px;
+    background-color: black;
+    opacity: 0.6;
+    width: 100%;
+    height: 100%;
+    padding-top: 200px;
```
Comment (spryor) [inline src/components/dashboardPanel/dashboardPanel.scss]: Remove the top padding.

File: src/components/deviceMap/deviceMap.js [hunk 6]
```diff
@@ -168,7 +168,7 @@ class DeviceMap extends Component {
   render() {
     const { BingMapKey } = this.props;
     return (
-      <DashboardPanel title={lang.DEVICELOCATION} indicator={this.props.showSpinner} className="map-container">
+      <DashboardPanel title={lang.DEVICELOCATION} initialIndicator={this.props.showInitialSpinner} indicator={this.props.showSpinner} className="map-container">
```
Comment (spryor) [inline src/components/deviceMap/deviceMap.js]: Rename `initialIndicator` to `showSpinner` or something. Whether the spinner is being shown at the initial load or afterward isn't really important to the flag.

Comment (swetha571) [reply src/components/deviceMap/deviceMap.js]: IntialIndicator is shown only on intila load and filter change.

File: src/components/deviceMap/deviceMap.js (hunk 2 above)
Comment (swetha571) [reply src/components/deviceMap/deviceMap.js]: Sure.

File: src/components/deviceMap/deviceMap.js [hunk 7]
```diff
@@ -168,7 +168,7 @@ class DeviceMap extends Component {
   render() {
     const { BingMapKey } = this.props;
     return (
-      <DashboardPanel title={lang.DEVICELOCATION} indicator={this.props.showSpinner} className="map-container">
+      <DashboardPanel title={lang.DEVICELOCATION} showSpinner={this.props.showInitialSpinner} indicator={this.props.showSpinner} className="map-container">
```
Comment (spryor) [inline src/components/deviceMap/deviceMap.js]: I don't see any difference between showInitialSpinner and showSpinner. The reason for showing the spinner doesn't matter, only whether or not to show it.

Comment (swetha571) [reply src/components/deviceMap/deviceMap.js]: showSpinner is for the refresh, placed in the headerpanel and showInitialSpinner is for the on load and filter change, placed in the panelContents.

Comment (spryor) [reply src/components/deviceMap/deviceMap.js]: Ok, these names need to be changed then to better indicate what they do. There is no obvious difference between `showSpinner` and `indicator`. They appear to be the same. Perhaps `showHeaderSpinner` and `showContentSpinner`.

**Pull request merged**